src/js/component.js

```diff
@@ -82,1182 +90,1103 @@ var Component = CoreObject.extend({
       this.enableTouchActivity();
     }
   }
-});
 
-/**
- * Dispose of the component and all child components
- */
-Component.prototype.dispose = function(){
-  this.trigger({ type: 'dispose', 'bubbles': false });
+  // Temp for ES6 class transition, remove before 5.0
+  init() {
+    // console.log('init called on Component');
+    Component.apply(this, arguments);
+  }
 
-  // Dispose all children.
-  if (this.children_) {
-    for (var i = this.children_.length - 1; i >= 0; i--) {
-      if (this.children_[i].dispose) {
-        this.children_[i].dispose();
+  /**
+   * Dispose of the component and all child components
+   */
+  dispose() {
+    this.trigger({ type: 'dispose', 'bubbles': false });
+
+    // Dispose all children.
+    if (this.children_) {
+      for (var i = this.children_.length - 1; i >= 0; i--) {
```

Comment:
if we are making es5-shim a requirement, I would love to turn all for-loops into iterators.